DQMOffline/Trigger/plugins/BTagAndProbe.cc

```diff
@@ -0,0 +1,1277 @@
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/Registry.h"
+#include "FWCore/Utilities/interface/transform.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "CommonTools/Utils/interface/StringCutObjectSelector.h"
+#include "CommonTools/TriggerUtils/interface/GenericTriggerEventFlag.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Common/interface/ValueMap.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/MuonReco/interface/Muon.h"
+#include "DataFormats/MuonReco/interface/MuonFwd.h"
+#include "DataFormats/GsfTrackReco/interface/GsfTrack.h"
+#include "DataFormats/EgammaCandidates/interface/GsfElectronFwd.h"
+#include "DataFormats/EgammaCandidates/interface/Photon.h"
+#include "DataFormats/EgammaCandidates/interface/PhotonFwd.h"
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/JetReco/interface/PFJetCollection.h"
+#include "DataFormats/BTauReco/interface/JetTag.h"
+#include "DataFormats/METReco/interface/PFMET.h"
+#include "DataFormats/METReco/interface/PFMETCollection.h"
+#include "DQMOffline/Trigger/plugins/TriggerDQMBase.h"
+
+//Tagging variables
+#include "DataFormats/BTauReco/interface/ShallowTagInfo.h"
+//#include "DataFormats/BTauReco/interface/ChargedCandidateFeatures.h"
+//#include "DataFormats/BTauReco/interface/NeutralCandidateFeatures.h"
+//#include "DataFormats/BTauReco/interface/SecondaryVertexFeatures.h"
+//#include "DataFormats/BTauReco/interface/DeepFlavourFeatures.h"
+//#include "DataFormats/BTauReco/interface/DeepFlavourTagInfo.h"
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <map>
+
+class BTagAndProbe : public DQMEDAnalyzer, public TriggerDQMBase {
+public:
+  typedef dqm::reco::MonitorElement MonitorElement;
+  typedef dqm::reco::DQMStore DQMStore;
+
+  BTagAndProbe(const edm::ParameterSet&);
+  ~BTagAndProbe() throw() override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+protected:
+  void bookHistograms(DQMStore::IBooker&, edm::Run const&, edm::EventSetup const&) override;
+  void analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup) override;
+
+  struct JetRefCompare {
+    inline bool operator()(const edm::RefToBase<reco::Jet>& j1, const edm::RefToBase<reco::Jet>& j2) const {
+      return (j1.id() < j2.id()) || ((j1.id() == j2.id()) && (j1.key() < j2.key()));
+    }
+  };
+  typedef std::map<edm::RefToBase<reco::Jet>, float, JetRefCompare> JetTagMap;
+
+private:
+  const std::string folderName_;
+
+  const bool requireValidHLTPaths_;
+  bool hltPathsAreValid_;
+
+  edm::EDGetTokenT<reco::VertexCollection> vtxToken_;
```

Comment:
These can all be `const`
The same for all other const class members assigned in the class initialization list